### PR TITLE
🌱 Update hetznercluster_types.go

### PR DESCRIPTION
**Pull request by Sayanta66**
**What this PR does / why we need it**:

Added spec descriptions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [x] include documentation
- [ ] add unit tests

### CODE REVIEW

File: api/v1beta1/hetznercluster_types.go [hunk 1]
```diff
@@ -27,18 +27,18 @@ const (
...
 	// AllowEmptyControlPlaneAddressAnnotation allows HetznerCluster Webhook
-	// to skip some validation steps for external managed controle planes.
+	// to skip some validation steps for externally managed control planes.
 	AllowEmptyControlPlaneAddressAnnotation = "capi.syself.com/allow-empty-control-plane-address"
 )
 
 // HetznerClusterSpec defines the desired state of HetznerCluster.
 type HetznerClusterSpec struct {
-	// HCloudNetworkSpec defines the Network for Hetzner Cloud. If left empty no private Network is configured.
+	// HCloudNetworkSpec defines details about the Private Network for Hetzner Cloud. If left empty, no private Network is configured.
```
Comment (janiskemper) [inline api/v1beta1/hetznercluster_types.go:36]: I guess we should rather write it without any capital letters to have consistency -> private network

File: api/v1beta1/hetznercluster_types.go [hunk 2]
```diff
@@ -27,18 +27,18 @@ const (
...
 	// +optional
 	HCloudNetwork HCloudNetworkSpec `json:"hcloudNetwork"`
 
 	// ControlPlaneRegion consists of a list of HCloud Regions (fsn, nbg, hel). Because HCloud Networks
-	// have a very low latency we could assume in some use-cases that a region is behaving like a zone
+	// have a very low latency we could assume in some use cases that a region is behaving like a zone
```
Comment (janiskemper) [inline api/v1beta1/hetznercluster_types.go:41]: "." at the end

File: api/v1beta1/hetznercluster_types.go [hunk 3]
```diff
@@ -48,7 +48,7 @@ type HetznerClusterSpec struct {
 	// +optional
 	ControlPlaneEndpoint *clusterv1.APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
 
-	// ControlPlaneLoadBalancer is optional configuration for customizing control plane behavior. Naming convention is from upstream cluster-api project.
+	// ControlPlaneLoadBalancer is an optional configuration for customizing control plane behavior. The naming convention is from the upstream cluster-api project.
```
Comment (janiskemper) [inline api/v1beta1/hetznercluster_types.go:51]: Cluster API

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: api/v1beta1/hetznercluster_types.go [hunk 4]
```diff
@@ -27,18 +27,18 @@ const (
...
 	// AllowEmptyControlPlaneAddressAnnotation allows HetznerCluster Webhook
-	// to skip some validation steps for external managed controle planes.
+	// to skip some validation steps for externally managed control planes.
 	AllowEmptyControlPlaneAddressAnnotation = "capi.syself.com/allow-empty-control-plane-address"
 )
 
 // HetznerClusterSpec defines the desired state of HetznerCluster.
 type HetznerClusterSpec struct {
-	// HCloudNetworkSpec defines the Network for Hetzner Cloud. If left empty no private Network is configured.
+	// HCloudNetworkSpec defines details about the private Network for Hetzner Cloud. If left empty, no private Network is configured.
```
Comment (janiskemper) [inline api/v1beta1/hetznercluster_types.go:36]:
```suggestion
	// HCloudNetwork defines details about the private Network for Hetzner Cloud. If left empty, no private Network is configured.
```

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Review by janiskemper — DISMISSED** [review]

File: api/v1beta1/hetznercluster_types.go [hunk 5]
```diff
@@ -48,7 +48,7 @@ type HetznerClusterSpec struct {
 	// +optional
 	ControlPlaneEndpoint *clusterv1.APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
 
-	// ControlPlaneLoadBalancer is optional configuration for customizing control plane behavior. Naming convention is from upstream cluster-api project.
+	// ControlPlaneLoadBalancer is an optional configuration for customizing control plane behavior. The naming convention is from the upstream Cluster API project.
```
Comment (kranurag7) [inline api/v1beta1/hetznercluster_types.go:51]: // cc @janiskemper not a blocker but I think upstream only mandates `spec.controlPlaneEndpoint` 

Ref: https://main.cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure#infracluster-resources 

I think we can omit the part of naming convention given I was not able to find any upstream reference.

Comment (janiskemper) [reply api/v1beta1/hetznercluster_types.go:51]:
```suggestion
	// ControlPlaneLoadBalancer is an optional configuration for customizing control plane behavior.
```

I agree, @Sayanta66 plz remove the second sentence

**Review by kranurag7 — CHANGES_REQUESTED** [review]
The description field doesn't match what we have defined on types as comments. 
Can you please run `make generate-manifests` again and push the same. 

Rest, looks good to me.

**Review by kranurag7 — APPROVED** [review]

**Pull request merged**